service/header/p2p_exchange.go

```diff
@@ -41,13 +41,12 @@ func NewP2PExchange(host host.Host, peer *peer.AddrInfo, store Store) *P2PExchan
 		trustedPeer: peer,
 		connected:   make(chan struct{}),
 	}
-	host.SetStreamHandler(exchangeProtocolID, ex.requestHandler)
 	ex.host.Network().Notify(&network.NotifyBundle{ConnectedF: ex.Connected})
 	return ex
 }
 
 func (ex *P2PExchange) Start(ctx context.Context) error {
-	log.Info("starting p2p exchange")
+	log.Info("p2p: starting p2p exchange")
```

Comment:
we might need to consider having sub-packages under `header` for exchange and subcribers

Reply:
hmmmmmm yes. Maybe write an issue for it

Reply:
https://github.com/celestiaorg/celestia-node/issues/322